app/jobs/ingest_job.rb

```diff
@@ -48,9 +48,7 @@ def perform(model_params:, signed_ids:, background_job_result:,
     # Create workflow destroys existing steps if called again, so need to check if already created.
     Workflow.create_unless_exists(druid, 'registrationWF', version: 1, priority: priority)
 
-    StageFiles.stage(signed_ids, druid) do
-      Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow
-    end
+    stage_files(signed_ids: signed_ids, globus_ids: globus_ids, druid: druid, priority: priority) if start_workflow
```

Comment:
Looking at this delta, beforehand, we would stage files all the time and create the workflow if `start_workflow` was truthy. Now it looks like we only stage files if `start_workflow` is truthy. Did I get that right? Should that behavior change be in the refactoring? (Or is it harmless?)

Reply:
Related to ☝🏻, I wonder if it's worth inlining the code so that the delta here has:

```ruby
StageBlobs.stage(signed_ids, druid) do
 Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow
end

StageGlobus.stage(globus_ids, druid) do
  Workflow.create_unless_exists(druid, 'accessionWF', version: 1, priority: priority) if start_workflow
end
```

❓

Reply:
Works for me :-) I'll update.

Reply:
@mjgiarlo actually - I the problem I ran into with that is that we'll always create 2 workflows now... so I wonder if it would be better to add a `Workflow.create...` if there are no signed_ids or globus_ids. Thoughts?

Reply:
Hmmm, shouldn't this logic take care of not creating a duplicate workflow? https://github.com/sul-dlss/sdr-api/blob/main/app/services/workflow.rb#L11